docker/docker_client.go

```diff
@@ -42,17 +42,17 @@ type dockerClient struct {
 	wwwAuthenticate string // Cache of a value set by ping() if scheme is not empty
 	scheme          string // Cache of a value returned by a successful ping() if not empty
 	client          *http.Client
+	signatureBase   signatureStorageBase
 }
 
 // newDockerClient returns a new dockerClient instance for refHostname (a host a specified in the Docker image reference, not canonicalized to dockerRegistry)
-func newDockerClient(ctx *types.SystemContext, refHostname string) (*dockerClient, error) {
-	var registry string
-	if refHostname == dockerHostname {
+// forWriting specifies whether the client will be used for "write" access (in particular passed to lookaside.go:toplevelFromSection)
+func newDockerClient(ctx *types.SystemContext, ref dockerReference, forWriting bool) (*dockerClient, error) {
```

Comment:
can we name this just `write bool` and document that it'll be used _for writing_? This is not in the Go idioms where variables,arguments and such are just small names. Maybe `mode` or `mask` and have other values to maybe be future proof?